common-rest/src/main/java/org/uniprot/api/rest/service/BasicSearchService.java

```diff
@@ -100,13 +177,37 @@ public SolrRequest createSolrRequest(
         }
         requestBuilder.query(requestedQuery);
 
-        requestBuilder.addSort(solrSortClause.getSort(request.getSort(), hasScore));
+        if (solrSortClause != null) {
+            requestBuilder.addSort(solrSortClause.getSort(request.getSort(), hasScore));
+        }
 
-        if (request.hasFacets()) {
+        if (includeFacets && request.hasFacets()) {
             requestBuilder.facets(request.getFacetList());
             requestBuilder.facetConfig(facetConfig);
         }
 
+        // set the solr batch size if passed else use DEFAULT_SOLR_BATCH_SIZE(25)
+        requestBuilder.rows(this.solrBatchSize.orElse(DEFAULT_SOLR_BATCH_SIZE));
```

Comment:
can we put this in an `else` branch of the below `if`, to make a bit easier to read?